File: src/xml/sax/java/XMLParser.java
import javax.xml.parsers.*;
import org.xml.sax.*;
import org.xml.sax.helpers.*;

import java.util.*;
import java.io.*;

public class XMLParser extends DefaultHandler {

	private static int currentNode = 0;
	private static int deepestNode = -1;

	public static void main(String[] args) throws Exception {
		SAXParserFactory factory = SAXParserFactory.newInstance();
		factory.setNamespaceAware(true);
		SAXParser parser = factory.newSAXParser();	

		XMLReader xmlReader = parser.getXMLReader();
		xmlReader.setContentHandler(new XMLParser());
		
		xmlReader.parse(args[0]);
		System.out.printf("Deepest node: %d\n", deepestNode);
	}
	
	@Override
	public void startElement(String namespaceURI,
              	          		String localName,
             		            String qName, 
             		            Attributes atts) throws SAXException {
		++currentNode;
		deepestNode = Math.max(deepestNode, currentNode);
	}
	@Override
	public void endElement(String namespaceURI,
                     		    String localName,
                 		        String qName) throws SAXException {
		--currentNode;
	}			

	@Override
	public void startDocument() throws SAXException {}

	@Override
	public void endDocument() throws SAXException {}

}

File: src/xml/sax/csharp/XMLParser.cs
using System;
using System.Xml;



public class XMLParser
{
	private static int deepestNode = -1;
	private static int currentNode = 0;
	

	public static void Main(string[] args){
		new XMLParser(args[0]);

		Console.WriteLine("Deepest Node: " + deepestNode);

	}	

	public XMLParser(string input){
		using(XmlReader reader = XmlReader.Create(input))
		{
			while(reader.Read()){
				switch(reader.NodeType)
				{
					case XmlNodeType.Element:
						if(!reader.IsEmptyElement)
							startElement();
						break;
					case XmlNodeType.EndElement:
						endElement();
						break;
				}
			}
		}


	}

	private void startElement(){
		++currentNode;
		deepestNode = Math.Max(currentNode, deepestNode);
	}

	private void endElement(){
		--currentNode;
	}
}